### DISCUSSION TITLE
HaveIBeenPwned Password Cracker

### DISCUSSION THREAD

**Discussion by lulumamba** [community/community#22639] [New to GitHub]
<p>Hello, I use Github only occasionally to write issues and the like. I have just heard about the partnership between Github and HaveIBeenPwned. HaveIBeenPwned has been known to me for a long time, and not positive.</p>
<p>What is Github doing with HaveIBeenPwned? Github promises that it can only see the passwords of the users hashed. With HaveIBeenPwned Github cracks the passwords of the users. OK, not all, but all that exists in any of the lists provided by HaveIBeenPwned. 517 million records.</p>
<p>The problem: Github does not crack the passwords of users in connection with the email addresses, as they are in the lists of HaveIBeenPwned. Github cracks the passwords of the user independently of the email addresses.</p>
<p>How it works? The hashed passwords in the HaveIBeenPwned lists are hopped using the same hash method as Gashub hashed. Then the hashed passwords are compared and if there is a hit, the unashashed password is assigned to the Github account. Hereby just independent of the email address and possible a new email address. HaveIBeenPwned thus extends its offer.</p>
<p>HaveIBeenPwned made the breakthrough. With HaveIBeenPwned, however, quite different things are possible / on the run.</p>
<p>I trust HaveIBeenPwned ZERO! (my experience and reasonable sense). And that should be any reasonable internet user.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>As stated in our <a href="https://blog.github.com/2018-07-31-new-improvements-and-best-practices-for-account-security-and-recoverability/" rel="nofollow noopener">recent blog post</a>, <a href="http://HaveIBeenPwned.com" rel="nofollow noopener">HaveIBeenPwned.com</a> has made <a href="https://haveibeenpwned.com/Passwords" rel="nofollow noopener">a list of passwords that have been compromised in various data breaches</a> available for download. These are passwords that people attacking accounts will be more likely to use than passwords generated random brute force methods because most people use the same password or some small set of passwords for all their accounts. Because of this, it is advisable that people should not use these compromised passwords if they want their accounts to be, and remain, secure.</p>
<p>In order to help people keep their GitHub accounts secure, we now check the password you provide to us against the list of compromised passwords and let you know if the password you provide is present in that list. We do not store the password you provide to us anywhere, only the <a href="https://en.wikipedia.org/wiki/Bcrypt" rel="nofollow noopener">bcrypt hash</a> of the password.</p>
<p>GitHub does not “crack” your password. HaveIBeenPwned does not “crack” your password. Nothing is sent from GitHub to HaveIBeenPwned because GitHub uses a local copy of the password list that HaveIBeenPwned provides for download.</p>
<p>I hope that clarifies things.</p>

**Comment by alexkami27073-hub** [comment]
https://www.instagram.com/miss_pritilina6297?igsh=M25wM3Fic2UyMWpz

**Comment by alexkami27073-hub** [comment]
How to crack  password
